### PR TITLE
Add Apply Workflow End-to-End Verification Test Plan

### PR DESCRIPTION
**Pull request by Bhavna-Ramachandran**
## Summary
Fixes / Work for #[6146 ](https://github.com/HHS/simpler-grants-gov/issues/6146) 

## Changes proposed
- Created folder structure for Apply workflow test documentation: frontend/tests/e2e/apply/testplan/
- Added Apply Workflow End-to-End Verification test plan
- Documented testing scope, approach, environments, risks, and traceability for Apply workflow

## Context for reviewers
This PR introduces the first formal E2E verification test plan for the Apply workflow.
The document defines how happy path, negative, and edge-case scenarios will be validated across  BDD, and Playwright E2E test layers.

Reviewers should verify that:
- The scope accurately reflects current Apply workflow behavior
- Test coverage aligns with existing and planned E2E automation
- Risks, entry/exit criteria, and traceability are complete and clear

## Validation steps
- Confirm the test plan exists at: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md
- Review the document structure and ensure it clearly defines:
- Scope and objectives
- Test approach and environments
- Entry/exit criteria and risks
- Verify the folder structure supports future Apply workflow test plans and documentation.

### CODE REVIEW

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 1]
```diff
@@ -0,0 +1,132 @@
+# Apply Workflow End-to-End Verification plan
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: can we take a quick step back and talk about what you see as the file structure we'll have set up here? Is it something like
```
- frontend
  - tests
    - e2e
      - <feature name>
        - features
          - feature files
        - testplan
          - test plan markdown files
        - specs
          - playwright test spec files
        - ???
          - any other file types that I'm missing?
```

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+### 3.2. Out of Scope
+
+• Post-submission editing
+• Reviewer/Admin workflows
+• Deep search or dashboard functionality
+• Email notifications
+• Upload file size validations, format validations and virus checks
+• i18n translations, accessibility, or API validation
+• Mobile browser checks
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I think we can keep mobile checks in place. Our mobile experience is determined by viewport size alone at this point, so it should be testable in any scenario. While we have some acknowledged deficiencies in our mobile experience on the apply workflow, and some things that we do not intend to fully work at a small viewport size, it would be nice to track what is working and what isn't so that we can maintain the proper baseline of functionality

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+## 4. Test Objectives
+
+• Validate that users can login and access Search Funding Opportunity page.
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I wonder to what degree we want to keep search, login, and apply separate from the apply objectives? Search will need its own set of tests, and login will be a common theme across a number of features. I understand calling out successful login and search as pre-requisites for the apply workflow, but would like to make sure we think of them as features of their own as well.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+• Does not cover i18n, accessibility, or API validation
+• Location:
+o frontend/tests/e2e/apply/features/happy_path_apply.feature
+o frontend/tests/e2e/apply/negativecases_apply.feature
+o frontend/tests/e2e/apply/edgecases_apply.feature
+
+### 5.2. End-to-End Tests (Playwright)
+
+• Validates full user experience: Start → Fill → Upload → Review → Submit
+• Includes autosave, negative/edge case handling, network errors, browser compatibility checks
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: we don't have autosave functionality at this point, would remove that mention

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+• Location:
+o frontend/tests/e2e/apply/happypath.specs.ts
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: can we decide on standard file naming convention for all e2e file names? I am fine with either camelCase, snake_case, or whatever-it-is-called-when-you-use-dashes, but we should be consistent. The rest of the frontend app uses camelCase, but there are examples of each across the code base

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+• Verify optional form selection works as expected.
+• Confirm documents upload correctly.
+• Ensure UI components behave consistently across workflow states.
+• Validate submission confirmation page loads with expected content.
+
+## 5. Test Approach
+
+Testing is divided into the following levels:
+
+### 5.1. BDD (Gherkin) Feature Tests
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Can we remove mentions of Gherkin until we figure out what tooling we'll be using?

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+• Location:
+o frontend/tests/e2e/apply/happypath.specs.ts
+o frontend/tests/e2e/apply/negativecases.specs.ts
+o frontend/tests/e2e/apply/edgecases.specs.ts
+
+## 6. Test Environment
+
+• Staging - QA verification in a production-like environment - Full Playwright end-to-end tests covering happy path, negative and edge cases, optional forms, attachment uploads, and submission confirmation
+o URL:- https://staging.simpler.grants.gov/
+• Production - Happy Path E2E Check to ensure submission works and confirmation page displays correctly
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Do you want to run automated e2e tests against production as well? I don't think that's in scope for right now, but we could talk about that for the future.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+o URL:- https://simpler.grants.gov/
+
+## 7. Entry & Exit Criteria
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: what does this section mean?

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 1 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: You are right. The structure you outlined is what I had in mind. Conceptually:
frontend/tests/e2e/
  |--<feature name>/                  # Feature-level folder (e.g., apply, search)
  |-----|--- <artifact name>/      # Artifact folder (e.g., features, testplan, steps, specs)
  |--------------|--- <files>/        # Actual files inside the artifact folder

With this structure in place, the E2E folder in our current GitHub repo would appear as follows:
frontend/tests/e2e/
  |---apply/
  |----|--- features/    (BDD-style feature files / high-level workflow definitions)
  |----|--- testplan/    (test plan / verification markdown files - living documents that evolve as functionality grows)
  |----|--- steps/       (optional - step definition files if we adopt a step-based / BDD approach)
  |----|--- specs/       (optional / proposed - Playwright test spec files)
  |----|--- (others as needed, e.g. helpers, test data)
  |
  |---search/
  |----|--- features/    (BDD-style feature files / high-level workflow definitions)
  |----|--- testplan/    (test plan / verification markdown files - living documents that evolve as functionality grows)
  |----|--- steps/       (optional - step definition files if we adopt a step-based / BDD approach)
  |----|--- specs/       (optional / proposed - Playwright test spec files)
  |----|--- (others as needed, e.g. helpers, test data)

Currently, I see that Playwright spec files are maintained directly under the <feature-name> folder, so I followed the same pattern in Login flow PR for consistency. That said, I like the idea (as in your comment above) of moving spec files into a dedicated folder, as it would make the structure more organized and easier to scale as we add more tests for each functionality under the feature. This is a proposed approach, I am open to align with the team’s preferred direction.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 3 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Agreed. Login and Search are their own features. My intent in listing them here is to verify the core Apply objectives as a complete real world end-to-end user flow, which requires certain preconditions to be satisfied - specifically, that a user can log in and search for a funding opportunity. That’s why these steps appear in the Preconditions section. 

Currently, the happy path feature covers the full E2E workflow, including Login, Search, application creation, and submission. In addition, we would have separate tests for apply specific negative cases and edge cases to ensure coverage beyond the core happy path. Please let me know your thoughts on this.

Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Discussed with Doug and keeping Login,Search in preconditions and removing their mention from other sections as the primary scope of verification is Apply workflow.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 6 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Acknowledged

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 4 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Acknowledged

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 8 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Discussed with Doug

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 7 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Yes just the E2E happy path test post latest code deployment to Prod. To make sure that the code changes did not break anything in Prod from user experience perspective.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 5 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Thanks for pointing this out! I agree - we should standardize E2E file names. I noticed that existing folders were using camelCase, we can use the same for naming files to be consistent. We can create a quick guideline and update the existing files over time as part of maintenance task. Let me know your thoughts.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 7 above)
Comment (doug-s-nava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: That's a good idea. I've started a list of things we need to do, and added that to the list so we don't lose track https://docs.google.com/spreadsheets/d/1UwOYG21f2YjHLBCrM1Nhck8WAX40biBDCN4UiV9nPuo/edit?usp=sharing

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 1 above)
Comment (doug-s-nava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: This sounds good to me - in line with the other comment about file names, can this be `testPlan` instead of `testplan`?

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 5 above)
Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: From experience I'd recommend using dashes.  E2E descriptive name can get long, and having dashes rather than camel case make them much easier to read.

Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I'd like more descriptive file names rather than things like `happypath`  and `edgecases` as it isn't clear what they are actually testing.  Even in the table below you have to specify what each happy path and each edgecase is in a separate column.  I'd rather that embedded in the name so when something fails you know exactly what went wrong.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 7 above)
Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: How are we managing different suites of test cases?  For instance we'll want to run all of them against localhost/ci, most against staging, and a subset smoke set against production.

Note that our test competitions are different in all three of those environments, so we'll want to find a way to take that into account/work on creating some standard competitions across all environments.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+Entry Criteria:
```
Comment (ErinPattisonNava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Entry Criteria for writing tests?  Or executing them?

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+• No blockers or high severity defects open
+• E2E happy path workflow passes reliably
+
+## 8. Risks
+
+### API instability
+
+The backend APIs that the Apply workflow depends on may be slow, unreliable, or return errors. This can cause tests to fail or the application to behave unpredictably during submission, form saving, or document uploads.
+
+### Large file uploads
```
Comment (ErinPattisonNava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Not sure how this is a risk for writing tests.  Feels like this could be its own performance suite of tests.  Not sure if we do pen testing here either and this might be better covered under that.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+• The user can access Search funding opportunities page.
+• The user can search for and select a specific Funding Opportunity.
+• Required test data (organization or individual profiles, documents) is available.
+• The system is available and stable.
+
+## 3. Scope
+
+### 3.1. In Scope
+
+This Test Plan covers functional and non-functional aspects of the Apply Workflow:
```
Comment (ErinPattisonNava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I don't understand the non-functional aspect?

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+## 1. Purpose
+
+The purpose of this Test Plan is to define the testing strategy, scope, roles, environments, schedule, preconditions, and deliverables for validating the Apply Workflow within Simpler Grants.
+This ensures that Organization and Individual users can successfully start an application, complete required and conditionally required forms, upload documents, review, and submit a grant application without errors.
+
+## 2. Preconditions
+
+Before executing Apply workflow tests, the following must be satisfied:
+• The user is able to login successfully with all required roles to create an application.
```
Comment (ErinPattisonNava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: These dots don't format correctly when you view the file in github.  Can you switch to using `*` so it becomes a bulleted list?

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+
+| Workflow Step                         | Test Type | Location (Exact Path)                                                                                                                               | Notes                                                                                    |
+| ------------------------------------- | --------- | --------------------------------------------------------------------------------------------------------------------------------------------------- | ---------------------------------------------------------------------------------------- |
+| User login with necessary roles       | E2E       | `frontend/tests/e2e/apply/happypath.specs.ts`<br>`frontend/tests/e2e/apply/negativecases.specs.ts`                                                  | Happy: successful login<br>Negative: invalid credentials, missing roles                  |
+| Search and select Funding Opportunity | E2E       | `frontend/tests/e2e/apply/happypath.specs.ts`<br>`frontend/tests/e2e/apply/negativecases.specs.ts`                                                  | Happy: valid opportunity<br>Negative: invalid or non-existent search                     |
```
Comment (ErinPattisonNava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: These are search tests, not apply tests.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+
+### Browser inconsistencies
+
+The Apply workflow may behave differently across browsers (Chrome, Firefox, Edge, Safari, Mobile browsers) due to rendering, JavaScript execution, or CSS differences. This can lead to UI misalignment, broken functionality, or inconsistent user experience.
+
+### Multi-step form state loss
+
+The workflow involves multiple steps and forms. There’s a risk that form data could be lost if the user navigates away or a network request fails. This can disrupt the user experience and lead to submission errors.
+
+## 9. Traceability Table
```
Comment (ErinPattisonNava) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: This feels fragmented into parts that are too small, which I'm concerned about from a performance perspective.  It is difficult to right-size e2e tests to balance setup/tear down costs with intermittent failures, but this feels off.  I'd like to view e2e tests as covering creating an application -> submission to make sure we are getting a good quality guarantee.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 5 above)
Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I think we're also going to probably want to do form-by-form testing.  So I'd suggest we have something similar to 

```
frontend/tests/e2e/apply/form-cd511/add-minimal-fields-succes.test.js
frontend/tests/e2e/apply/form-cd511/upload-two-attachments-success.test.js
frontend/tests/e2e/apply/form-cd511/organization-owned-requirements-success.test.js
```

Err, these aren't the use cases for form CD511, but you can see how much more readable and meaningful the filenames are.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 14 above)
Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I also don't think we should have 100% coverage.  e2e tests are expensive, so we'll want to carefully pick what we invest in.  For instance, I suspect that most of what you have below as negative cases (which would be named more descriptively) aren't worth our time/resources.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 15]
```diff
@@ -0,0 +1,118 @@
...
+• Review application
+• Submission flow and confirmation page
+• End-to-end workflows in a real environment
+• Mobile browser checks
+
+### 3.2. Out of Scope
+
+• Post-submission editing
+• Reviewer/Admin workflows
+• Deep search or dashboard functionality
```
Comment (chris-kuryak) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: @Bhavna-Ramachandran I'm unsure what Deep Search or Dashboard functionality is referring to, can you clarify or give an example.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 16]
```diff
@@ -0,0 +1,118 @@
...
+• Start a new application for Organization and Individual users
+• Save and update application data
+• Required and conditionally required form validations
+• Optional form selection for submission
+• Document upload via forms and Attachments section
+• Navigation between steps
+• Review application
+• Submission flow and confirmation page
+• End-to-end workflows in a real environment
+• Mobile browser checks
```
Comment (chris-kuryak) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Note that we are not currently optimizing (or checking) for mobile compatibility on our Apply workflow. We are optimizing for desktop experience. Therefore, I don't think this needs to be in-scope for now, assuming we would optimize for mobile later in our development.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 17]
```diff
@@ -0,0 +1,118 @@
...
+### 3.1. In Scope
+
+This Test Plan covers functional and non-functional aspects of the Apply Workflow:
+• Start a new application for Organization and Individual users
+• Save and update application data
+• Required and conditionally required form validations
+• Optional form selection for submission
+• Document upload via forms and Attachments section
+• Navigation between steps
+• Review application
```
Comment (chris-kuryak) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: There is not currently a feature around "reviewing an application". What is this referring to?

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md [hunk 18]
```diff
@@ -0,0 +1,118 @@
...
+• Submission flow and confirmation page
+• End-to-end workflows in a real environment
+• Mobile browser checks
+
```
Comment (chris-kuryak) [inline frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: After successfully submitting an application, there are some backend processes that take place to create a Zip file of the submission package: 
- PDFs are created from the forms 
- XML is generated from the forms
- Attachments are added

Will these tests also be checking to ensure that process is completed as expected?

Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: The zip file/submission package is created asynchronously once an hour. We won't want to hold up the e2e tests for that long, so these aren't candidates to be tested through the e2e web-based framework.  Those are better suited to unittests and monitoring.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 7 above)
Comment (doug-s-nava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: @ErinPattisonNava beginning to figure that out is ticketed here, though the work of actually tagging tests and determining which groups to run when and where is maybe not ticketed yet? https://github.com/HHS/simpler-grants-gov/issues/7307

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 5 above)
Comment (doug-s-nava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: maybe a short meeting about this is in order?

Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Thank you @doug-s-nava  and @ErinPattisonNava ! We can have a quick meeting to help us align on a standard naming convention, finalize filenames, and choose a style for separators—whether camelCase, snake_case, or kebab-case.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 9 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: The Entry Criteria here refers to executing the E2E tests, not writing them - conditions that need to be met before running the tests (e.g., stable test environment set up, data availability...)

The Exit Criteria define the conditions that ensure the deployment is stable and error-free, so that no rollback is needed.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 10 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: This test plan and all its sections focus on executing the E2E tests, not writing them. The test scripts themselves remain the same regardless of file size; large uploads are only noted as a risk that could affect test reliability.

You’re right that very large file uploads or performance extremes could be part of a dedicated performance or penetration test suite, but here we’re noting them as potential risks to E2E test reliability.

Please let me know if you’d like me to include performance testing(under 'In Scope' section) in this plan, or keep it separate. In my previous projects, performance, capacity, stress, and accessibility tests were often part of the overall E2E plan. Maybe we can add this as discussion point during our meeting.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 11 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Functional = what the user can do / system features
Non-functional = how the system behaves while performing those actions. One common example for this would be : How the workflow works across different browsers and viewports. 

Few other examples in this workflow are listed below: 
Functional | Non-Functional
Attachment Uploads| Files appear correctly in the attachment table
Application Submission| Confirmation page loads correctly

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 12 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Acknowledged.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 13 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Thank you @ErinPattisonNava . Agreed - we don’t need invalid credentials for negative login, and we don’t need any negative cases for search here.  I will update this. However, we do need the “missing roles” case for negative login, since it tests whether the workflow allows a user to apply without the necessary roles.

As discussed with Doug (adding for context), for the happy path we need to include Login and Search. Although Login and Search are their own features, my intent in listing them here is to verify the core Apply objectives as a complete real world end-to-end user flow, which requires certain preconditions to be satisfied - specifically, that a user can log in and search for a funding opportunity. That’s why these steps appear in the Preconditions section (not part of the test objectives or In Scope section).

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 14 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I want to clarify that these aren’t fragments - the whole flow will be defined in single test script file for each case (happypath,negative,edge). The traceability table is meant to show and ensure that the full real-world Apply workflow is validated (like a checklist), indicating which files cover each part of the flow, along with the test type. All happy path, negative cases, and edge cases listed are E2E tests.

I agree that 100% coverage isn’t necessary for E2E tests. My understanding is that we are currently focusing on the happy path flow, and negative/edge cases can be prioritized later based on risk and value. This structure gives us the flexibility to identify and focus on the most critical scenarios.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 15 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Sure. “Deep search” is currently out of scope for the Apply workflow E2E tests and refers to validation of advanced search features.

A possible real-world deep search scenario would be: a user logs in, searches using multiple filters and keywords, selects a funding opportunity, starts an application, completes forms, uploads documents, and submits.

We could consider including deep search in the Apply workflow in future test coverage if validating advanced search is needed as part of the end-to-end user journey.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 16 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Acknowledged. I will move mobile browser checks as out of scope for now and we can revisit this later.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 17 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: By “review application”, I’m referring to the test validation step where assertions check - or in the real world, where users check - the completed application for errors or missing information before submission. This ensures that all required and conditionally required fields are filled correctly. It’s not a separate system feature, but a validation within the end-to-end workflow to ensure data completeness and accuracy prior to submission.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 18 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: @ErinPattisonNava is correct. We can revisit PDF/ZIP validation in E2E tests when these artifacts become available in the UI as part of the workflow in the future.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 11 above)
Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: I don't agree that the confirmation page loading correctly is "non-functional", nor do I think that how the workflow works across different browsers and viewports would be part of e2e testing.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 14 above)
Comment (ErinPattisonNava) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: There are multiple happy path flows, so I'd like to move away from talking about one in singular.

In order to test apply for instance, we probably want to test if we can apply as an organization and apply as an individual.  In each of those tests, it would include the steps you have below of starting the application, filling out the forms, and submission.  If we split out starting the application, then filling out forms, the filling out forms will need to include the starting the application step.  So now we're running the same test in two places, which makes our e2es more brittle and expensive.  It also makes it more difficult to understand what our total coverage is.  That's what I mean by it feels fragmented, I think the tests below are too narrow on a given use case and there are better ways to look at what end to end is for a feature.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 5 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Discussed: kebab-case is the preferred style.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 16 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: Discussed: Keep Mobile checks in for now. They can be revisited later.

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 1 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: renamed as test-plan (style used kebab-case per discussion)

File: frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md (hunk 5 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/testplan/apply_workflow_e2e_verification_plan.md]: updated from 'happypath.specs.ts' to 'end-to-end-apply-workflow.specs.ts' for the generic flow

File: frontend/tests/e2e/apply/features/e2e-apply-workflow.feature [hunk 19]
```diff
@@ -0,0 +1,152 @@
+Feature: Happy Path – Apply Workflow
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/features/e2e-apply-workflow.feature:1]: `e2e` in the filename is redundant, and `workflow` is going to quickly become a word we want to avoid unless we are talking specifically about the workflow feature we are building. I am fine with leaving this title as it was, something like `happy-path.feature` or `basic-submission-path.feature`

File: frontend/tests/e2e/apply/features/e2e-apply-workflow.feature [hunk 20]
```diff
@@ -0,0 +1,152 @@
...
+
+# =================================================================
+# --- Shared Workflow Details ---
+# =================================================================
+
+# --- Search the Opportunity ---
+When the user goes to the search page
+And the user searches for the specific Funding Opportunity number
+Then the user scrolls to the search results table
+When the user clicks the Funding Opportunity number
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/features/e2e-apply-workflow.feature]: the title text of the opportunity is the link, not the number

```suggestion
When the user clicks the Funding Opportunity title
```

File: frontend/tests/e2e/apply/test-plan/e2e-apply-workflow-test-plan.md [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+### 3.1. In Scope
+
+This Test Plan covers functional and non\*functional aspects of the Apply Workflow:
+
+- Start a new application for Organization and Individual users
+- Save and update application data
+- Required and conditionally required form validations
+- Optional form selection for submission
+- Document upload via forms and Attachments section
+- Navigation between steps
```
Comment (doug-s-nava) [inline frontend/tests/e2e/apply/test-plan/e2e-apply-workflow-test-plan.md]: this should be removed, there's no idea of steps in our system

File: frontend/tests/e2e/apply/features/e2e-apply-workflow.feature (hunk 19 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/features/e2e-apply-workflow.feature:1]: Acknowledged. Will update it to 'happy-path.feature'

File: frontend/tests/e2e/apply/features/e2e-apply-workflow.feature (hunk 20 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/features/e2e-apply-workflow.feature]: Acknowledged.

File: frontend/tests/e2e/apply/test-plan/e2e-apply-workflow-test-plan.md (hunk 21 above)
Comment (Bhavna-Ramachandran) [reply frontend/tests/e2e/apply/test-plan/e2e-apply-workflow-test-plan.md]: Acknowledged.
